### PR TITLE
Security exporter

### PR DESCRIPTION
**Pull request by perryism**
We are using the secrets decrypter across many projects.  Moving it here would make sense. 

# Implementation
I use Open3 so I can stub

### CODE REVIEW

File: lib/biscuit.rb [hunk 1]
```diff
@@ -1,9 +1,13 @@
...
+require "open3"
+require "yaml"
 
 module Biscuit
   def self.run!(command)
-    result = `#{__dir__}/../bin/_biscuit #{command}`
-    raise(result.slice(0, 200)) unless $?.success?
-    result
+    stdout, stderr, status = Open3.capture3("#{__dir__}/../bin/_biscuit #{command}")
+    raise(stderr.slice(0, 200)) unless status == 0
```
Comment (nevinera) [inline lib/biscuit.rb]: A difference (which may be an improvement) - previously we raised with a slice of the *stderr* output, but you have changed it to a slice of the *stdout* output. That seems likely to be more correct, but I wanted to make sure you were aware, and had checked out the actual behavior around this.

Comment (nevinera) [reply lib/biscuit.rb]: `Open3.capture3` is a *way* better tool for this by the way, even without the stubbing benefits :-)

Comment (nevinera) [reply lib/biscuit.rb]: Would you mind making this into a named exception? (maybe `Biscuit::Error`)

**Review by nevinera — COMMENTED** [review]
You'll probably need to bump the version too: https://github.com/usertesting/biscuit/blob/master/lib/biscuit/version.rb

File: lib/biscuit.rb (hunk 1 above)
Comment (perryism) [reply lib/biscuit.rb]: @nevinera previously the result holds both stdout and stderr I believe. maybe it should be `Biscuit::Error.new("#{stdout} #{stderr}".slice(0, 200))` thoughts?

**Comment by perryism** [comment]
@nevinera good call on version bump.  since a new small feature is introduced,  I will bump it to `0.1.0`, sound good?

File: lib/biscuit.rb (hunk 1 above)
Comment (perryism) [reply lib/biscuit.rb]: never mind,  it is `stdout` only

**Comment by perryism** [comment]
I can't decide whether I should raise error with stderr , or stdout.   so I defer that to the user that i put both in the execution error object.

File: lib/biscuit/execution_error.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module Biscuit
+  class ExecutionError < StandardError
+    def initialize(stdout, stderr)
```
Comment (nevinera) [inline lib/biscuit/execution_error.rb]: I'd feel a lot better about this if you changed the `stdout` parameter to be optional - I've run into a lot of places in the past where it was really irritating that someone had changed the initialization signature of a subclass.

**Review by nevinera — APPROVED** [review]

**Review by suan — APPROVED** [review]

**Comment by bobziuchkovski** [comment]
I'm late to the party on this one, but I love it!  😁

**Pull request merged**